pageserver/src/tenant.rs

```diff
@@ -200,37 +207,44 @@ impl UninitializedTimeline<'_> {
     }
 
     /// Prepares timeline data by loading it from the basebackup archive.
-    pub fn import_basebackup_from_tar(
-        &self,
-        reader: impl std::io::Read,
+    pub async fn import_basebackup_from_tar(
+        self,
+        mut copyin_stream: &mut Pin<&mut impl Stream<Item = io::Result<Bytes>>>,
         base_lsn: Lsn,
-    ) -> anyhow::Result<()> {
+    ) -> anyhow::Result<Arc<Timeline>> {
         let raw_timeline = self.raw_timeline()?;
-        import_datadir::import_basebackup_from_tar(raw_timeline, reader, base_lsn).with_context(
-            || {
-                format!(
-                    "Failed to import basebackup for timeline {}/{}",
-                    self.owning_tenant.tenant_id, self.timeline_id
-                )
-            },
-        )?;
+
+        // import_basebackup_from_tar() is not async, mainly because the Tar crate
+        // it uses is not async. So we need to jump through some hoops:
+        // - convert the input from client connection to a synchronous Read
+        // - use block_in_place()
```

Comment:
I think our current policy is to use block_in_place if it's local disk IO, since we assume that it's fast.
But our block_in_place below will read from the SyncIoBridge'd client connection, i.e., network IO.

Hence, I think `spawn_blocking` would be more appropriate.

Reply:
This is not introduced by this PR. Here I just moved this code from one function to another. See the discussion in the original [PR](https://github.com/neondatabase/neon/pull/2376/files#r962137730)

`block_in_place` is similar in many ways to `spawn_blocking`. And it is different from just blocking current thread as we do with disk operations. We indeed block in place when doing disk operations, but we're not using `block_in_place` :)

Still I agree that `spawn_blocking` fits better. Block in place is still ok but rather for background runtime, not the one we use for request handling. So any objections to replacing it separetely?

Reply:
No, go ahead

Reply:
At some point I think we'll move to async disk access. 

I'll try to fit spawn_blocking here after merge of this PR